### ISSUE TITLE
Scalar, views and multi-table transactions

### ISSUE THREAD

**Issue by maxlambertini** [amberframework/granite#327]
Hello, 

I'm testing Granite and, while concise and good, there are three things IMHO it partially covers (blame me being a DBA...) 

* Multi-table transactions
* Literal queries support and mapping results to a single scalar, an array of scalars or an array of instances. 
* Supporting tables without a primary key (thus enabling querying on Views: it can be done, but you have to "fake" a PK by designating a member as a PK, even if it is neither a key nor primary :-) )  

Are these features planned? Because I worked in C# with a home-made ORM and while mapping rows to instances is all well and good, enabling some low-level operation some times can be the cleaner way to get to your goal.... (case in point: COUNT(*) operations).

**Comment by maxlambertini** [comment]
Ok, found the "Custom Select" part. It is nifty but I also have to define a Model for every query.

**Comment by maxlambertini** [comment]
Tested the Custom Select feature and it does not work as intended.

I modeled a class as per your example:

~~~ruby
class CustomView < Granite:Base
  adapter pg
  field articlebody : String
  field commentbody : String

  select_statement <<-SQL
    SELECT articles.articlebody, comments.commentbody
    FROM articles
    JOIN comments
    ON comments.articleid = articles.id
  SQL
end
~~~

This is my original class

~~~ruby
class EventCon < Granite::Base
  adapter sqlite3

  field con_name          : String
  field event_name        : String

  select_statement <<-SQL
  select con_name, event_name FROM eventcons
  SQL
end
~~~

Executing `EventCon.all.each do | ev | .... end` yields the following stack trace:

~~~
max@mrfinch:~/dev/crystal/testdb/democreto$ bin/democreto 
Unhandled exception: SQLite3::ResultSet#read returned a String. A (Int64 | Nil) was expected. (Exception)
  from lib/db/src/db/result_set.cr:0:9 in 'read'
  from src/democreto.cr:30:1 in 'set_attributes'
  from src/democreto.cr:30:1 in 'from_sql'
  from lib/granite/src/granite/querying.cr:35:17 in 'raw_all'
  from lib/granite/src/granite/querying.cr:50:30 in '->'
  from lib/granite/src/granite/collection.cr:255:3 in 'collection'
  from /usr/share/crystal/src/indexable.cr:266:5 in '__crystal_main'
  from /usr/share/crystal/src/crystal/main.cr:97:5 in 'main_user_code'
  from /usr/share/crystal/src/crystal/main.cr:86:7 in 'main'
  from /usr/share/crystal/src/crystal/main.cr:106:3 in 'main'
  from __libc_start_main
  from _start
  from ???
~~~

Specifying a dummy ID field in your query works:

~~~ruby
class EventCon < Granite::Base
  adapter sqlite3

  field con_name          : String
  field event_name        : String

  select_statement <<-SQL
  select 1 as id, con_name, event_name FROM eventcons
  SQL
end
~~~

**Comment by Blacksmoke16** [comment]
@maxlambertini It looks like it's also wanting to return the ID of the table as well.  If you do not specify a PK using the `primary` macro, it defaults to `Int64`.  So maybe try returning the ID as well as a workaround.  

I'm thinking its a bug in that the point of the `select_statement` is to only select what you want, this should not have to require the id/PK column.

**Comment by Blacksmoke16** [comment]
@maxlambertini This will be fixed in #346.  At least the bug related to your `select_statement`

```cr
class EventCon < Granite::Base
  connection sqlite
  table event_cons

  column id : Int64, primary: true
  column con_name : String
  column event_name : String

  select_statement <<-SQL
  select 1 as id, con_name, event_name FROM event_cons
  SQL
end

EventCon.migrator.drop_and_create

EventCon.create(con_name: "ConOne", event_name: "EventOne")
EventCon.create(con_name: "ConTwo", event_name: "EventTwo")
EventCon.create(con_name: "ConThree", event_name: "EventThree")

EventCon.all.each do |env|
  puts env.id
  puts env.event_name
end
# =>
1
EventOne
1
EventTwo
1
EventThree
```
